spring-amqp/src/main/java/org/springframework/amqp/support/converter/AbstractMessageConverter.java

```diff
@@ -50,26 +52,46 @@ protected boolean isCreateMessageIds() {
 	@Override
 	public final Message toMessage(Object object, MessageProperties messageProperties)
 			throws MessageConversionException {
+		return toMessage(object, messageProperties, null);
+	}
+
+	@Override
+	public final Message toMessage(Object object, MessageProperties messageProperties, @Nullable Type genericType)
+			throws MessageConversionException {
 		if (messageProperties == null) {
 			messageProperties = new MessageProperties();
 		}
-		Message message = createMessage(object, messageProperties);
+		Message message = createMessage(object, messageProperties, genericType);
 		messageProperties = message.getMessageProperties();
 		if (this.createMessageIds && messageProperties.getMessageId() == null) {
 			messageProperties.setMessageId(UUID.randomUUID().toString());
 		}
 		return message;
 	}
 
+	/**
+	 * Crate a message from the payload object and message properties provided. The message id will be added to the
+	 * properties if necessary later.
+	 *
+	 * @param object the payload.
+	 * @param messageProperties the message properties (headers).
+	 * @return a message.
+	 */
+	protected abstract Message createMessage(Object object, MessageProperties messageProperties);
+
 	/**
 	 * Crate a message from the payload object and message properties provided. The message id will be added to the
 	 * properties if necessary later.
 	 *
 	 * @param object the payload
 	 * @param messageProperties the message properties (headers)
+	 * @param genericType the type to convert from - used to populate type headers.
 	 * @return a message
+	 * @since 2.1
 	 */
-	protected abstract Message createMessage(Object object, MessageProperties messageProperties);
+	protected Message createMessage(Object object, MessageProperties messageProperties, @Nullable Type genericType) {
+		return createMessage(object, messageProperties);
+	}
 
 	@Override
 	public abstract Object fromMessage(Message message) throws MessageConversionException;
```

Comment:
Sounds like we don't this method declaration in this class.
The same meaning we get from the an interface contract.